### PR TITLE
Improve setting to support float and big decimal

### PR DESCRIPTION
**Pull request by sangnguyen1001**

### CODE REVIEW

File: rails-settings-cached.gemspec [hunk 1]
```diff
@@ -32,4 +32,5 @@ Gem::Specification.new do |s|
   s.add_development_dependency "simplecov"
   s.add_development_dependency "minitest"
   s.add_development_dependency "codecov"
+  s.add_development_dependency "byebug"
```
Comment (huacnlee) [inline rails-settings-cached.gemspec]: Move this line into Gemfile

File: test/models/setting.rb [hunk 2]
```diff
@@ -19,4 +19,6 @@ class Setting < RailsSettings::Base
     client_id: "the-client-id",
     client_secret: "the-client-secret",
   }, type: :string, readonly: true
+  field :float_item, type: :float, default: 7
```
Comment (huacnlee) [inline test/models/setting.rb]: `7` -> `7.12`

File: lib/rails-settings/base.rb [hunk 3]
```diff
@@ -43,7 +46,12 @@ def cache_key
         scope.join("/")
       end
 
+      def setting_names
+        @@setting_names
+      end
```
Comment (huacnlee) [inline lib/rails-settings/base.rb]: `setting_names` is not a good name

How about `keys`.

```rb
Setting.keys
# => ["foo", "bar"] 
```

File: rails-settings-cached.gemspec (hunk 1 above)
Comment (huacnlee) [reply rails-settings-cached.gemspec]: Remove this dependency.

Comment (phuongnd08) [reply rails-settings-cached.gemspec]: @huacnlee this is just a development dependency. it helps dev to debug effectively when improving the gem, no?

File: Gemfile.lock [hunk 4]
```diff
@@ -65,6 +65,7 @@ GEM
       zeitwerk (~> 2.1, >= 2.1.4)
     ast (2.4.0)
     builder (3.2.3)
+    byebug (11.1.1)
```
Comment (huacnlee) [inline Gemfile.lock]: Revert this file.

I will merge this PR

**Comment by sangnguyen1001** [comment]
new PR here https://github.com/huacnlee/rails-settings-cached/pull/187/commits

**Pull request closed without merging**